dll/win32/kernel32/client/handle.c

```diff
@@ -187,6 +221,145 @@ DuplicateHandle(IN HANDLE hSourceProcessHandle,
         return FALSE;
     }
 
+	// winsock socket handles code currently only dupes from self or inherited sockets from parent are implemented
```

Comment:
But.... does Windows have all that supplemental code here? I strongly doubt that.

Reply:
I doubt it too, but it has 'something' undocumented that makes this work. There are some MSDN docs suggesting it is done through WSHGetSocketInformation and WSHSetSocketInformation Helper DLL functions both of which are unimplemented in ReactOS. [MSDN WSHGetSocketInformation page](https://learn.microsoft.com/en-us/previous-versions/windows/hardware/network/ff565992(v=vs.85))

Reply:
I think that this would be worth investigating further. (Also note there exists a [WSPDuplicateSocket](https://learn.microsoft.com/en-us/previous-versions/windows/hardware/network/ff566282(v=vs.85)) function.)

Reply:
WSPDuplicateSocket was the first thing I looked at and it is used in the Inheritance PR [absolutely also a hack] which this PR depends on (in that PR it is called from CreateProcessInternalW). The problem with that function is that It can only work from the process owning the original socket and DuplicateHandle is more often called from the 'target' process. That's the problem here.. this is a cross process operation and there is currently no provision for handling that IPC I can find anywhere in ReactOS (apparently if the MSDN page is correct, Windows handles it somewhere in the helper DLL).